### PR TITLE
[NUI] Fix to calculate LinearLayout's MatchParent children correctly

### PR DESCRIPTION
**Pull request by jaehyun0cho**
The followings have been fixed to calculate LinearLayout's MatchParent
children correctly.

- MatchParent child layout is calculated only if LinearLayout has
  positive Specification or MatchParent.

- MatchParent child layout's margin is not shared with other MatchParent
  child layouts.
  e.g.
  LinearLayout has size 100.
  Child layout1 is MatchParent and its margin is 20.
  Child layout2 is MatchParent and its margin is 0.
  Then, child layout1's size is 30 and child layout2's size is 50.

- Setting Width/HeightSpecification has been removed because
  Width/HeightSpecification should preserve the value set by user.

- Typo has been fixed. e.g. width, height

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/public/Layouting/LinearLayout.cs [hunk 1]
```diff
@@ -325,70 +326,78 @@ private void MeasureHorizontal(MeasureSpecification widthMeasureSpec, MeasureSpe
...
+            float remainingWidth = widthSize - totalLength;
             float totalWeightLength = 0.0f;
 
             // 2ND PHASE:
             //
             // We measure all children whose width specification policy is MatchParent without weight.
             // After 2nd phase, all children's widths are calculated without considering weight.
             // And the widths of all weighted children are accumulated to calculate weighted width.
-            foreach (LayoutItem childLayout in IterateLayoutChildren())
+            if (isWidthExactly)
```
Comment (everLEEst) [inline src/Tizen.NUI/src/public/Layouting/LinearLayout.cs:342]: Could be better to print error about circular dependency when parent is wrapContent and child is MatchParent

File: src/Tizen.NUI/src/public/Layouting/LinearLayout.cs [hunk 2]
```diff
@@ -495,70 +505,78 @@ private void MeasureVertical(MeasureSpecification widthMeasureSpec, MeasureSpeci
...
+                } // 2ND PHASE foreach
+            }
 
             // 3RD PHASE:
             //
             // We measure all weighted children whose owner has weight greater than 0.
             // After 3rd phase, all weighted children has height which is proportional to their weights
             // in remaining height of parent.
-            if (totalWeight > 0)
+            if ((isHeightExactly) && (totalWeight > 0))
```
Comment (everLEEst) [inline src/Tizen.NUI/src/public/Layouting/LinearLayout.cs:574]: Could be better to print error about circular dependency when parent is wrapContent and child is MatchParent

Comment (jaehyun0cho) [reply src/Tizen.NUI/src/public/Layouting/LinearLayout.cs]: Thank you for your comment.
The error log has been added.

**Review by everLEEst — APPROVED** [review]
LGTM

**Pull request merged**